src/main/java/com/wavefront/WavefrontTraceLoader.java

```diff
@@ -74,4 +75,26 @@ void dumpStatistics() throws Exception {
       System.out.println(spanGenerator.getStatistics());
     }
   }
+
+  void realTimeSending() throws InterruptedException {
+    // Send spans to host.
+    Thread generator = new Thread(spanGenerator);
+    Thread sender = new Thread(spanSender);
+    generator.start();
+    sender.start();
+    // Waiting while generation completes.
+    generator.join();
+    // Inform sender that generation completes.
+    spanSender.stopSending();
+    // Wait while sender devastates the span queue.
+    sender.join();
+  }
+
+  void generateSpans() {
+    spanGenerator.generate();
+  }
+
+  void saveSpansToFile() throws Exception {
+    spanSender.saveToFile();
+  }
```

Comment:
I will suggest getting rid of these one-line functions.

Reply:
Agree!